x/distribution/client/cli/tx.go

```diff
@@ -34,6 +35,237 @@ const (
+	if chunkSize == 0 {
+		return newGenerateOrBroadcast(cliCtx, txBldr, msgs...)
+	}
+
+	// split messages into slices of length chunkSize
+	totalMessages := len(msgs)
+	for i := 0; i < len(msgs); i += chunkSize {
+
+		sliceEnd := i + chunkSize
+		if sliceEnd > totalMessages {
+			sliceEnd = totalMessages
+		}
+
+		msgChunk := msgs[i:sliceEnd]
+		if err := newGenerateOrBroadcast(cliCtx, txBldr, msgChunk...); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func NewWithdrawRewardsCmd(m codec.Marshaler, txg tx.Generator, ar tx.AccountRetriever) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "withdraw-rewards [validator-addr]",
+		Short: "Withdraw rewards from a given delegation address, and optionally withdraw validator commission if the delegation address given is a validator operator",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Withdraw rewards from a given delegation address,
+and optionally withdraw validator commission if the delegation address given is a validator operator.
+
+Example:
+$ %s tx distribution withdraw-rewards cosmosvaloper1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj --from mykey
+$ %s tx distribution withdraw-rewards cosmosvaloper1gghjut3ccd8ay0zduzj64hwre2fxs9ldmqhffj --from mykey --commission
+`,
+				version.ClientName, version.ClientName,
+			),
+		),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txf := tx.NewFactoryFromCLI(inBuf).
+				WithTxGenerator(txg).
+				WithAccountRetriever(ar)
+			cliCtx := context.NewCLIContextWithInputAndFrom(inBuf, args[0]).WithMarshaler(m)
+
+			delAddr := cliCtx.GetFromAddress()
+			valAddr, err := sdk.ValAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			msgs := []sdk.Msg{types.NewMsgWithdrawDelegatorReward(delAddr, valAddr)}
+			if viper.GetBool(flagCommission) {
+				msgs = append(msgs, types.NewMsgWithdrawValidatorCommission(valAddr))
+			}
+
+			for _, msg := range msgs {
+				if err := msg.ValidateBasic(); err != nil {
+					return err
+				}
+			}
+
+			return tx.GenerateOrBroadcastTx(cliCtx, txf, msgs...)
+		},
+	}
+	cmd.Flags().Bool(flagCommission, false, "also withdraw validator's commission")
+	return flags.PostCommands(cmd)[0]
+}
+
+func NewWithdrawAllRewardsCmd(m codec.Marshaler, txg tx.Generator, ar tx.AccountRetriever) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "withdraw-all-rewards",
+		Short: "withdraw all delegations rewards for a delegator",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Withdraw all rewards for a single delegator.
+
+Example:
+$ %s tx distribution withdraw-all-rewards --from mykey
+`,
+				version.ClientName,
+			),
+		),
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			inBuf := bufio.NewReader(cmd.InOrStdin())
+			txf := tx.NewFactoryFromCLI(inBuf).
+				WithTxGenerator(txg).
+				WithAccountRetriever(ar)
+			cliCtx := context.NewCLIContextWithInputAndFrom(inBuf, args[0]).WithMarshaler(m)
```

Comment:
Should be `NewCLIContextWithInput` as `args[0]` actually doesn't point to anything - this will probably cause an out of bounds panic.

Reply:
Thanks for the review @aaronc, I see there's same issue in multiple modules. I addressed them all in a seperate PR